workers/reporter/src/main/kotlin/reporter/OrtResultReporter.kt

```diff
@@ -21,13 +21,34 @@ package org.eclipse.apoapsis.ortserver.workers.reporter
 
 import java.io.File
 
-import org.ossreviewtoolkit.model.config.PluginConfiguration
 import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.plugins.api.OrtPlugin
+import org.ossreviewtoolkit.plugins.api.OrtPluginOption
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
 import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterFactory
 import org.ossreviewtoolkit.reporter.ReporterInput
 import org.ossreviewtoolkit.utils.common.packZip
 
-class OrtResultReporter : Reporter {
+data class OrtResultReporterConfig(
+    /**
+     * Whether to compress the generated file. If set to `true` (which is the default), a Zip archive is generated which
+     * contains the ORT result file.
+     */
+    @OrtPluginOption(defaultValue = "true")
+    val compressed: Boolean
+)
+
+@OrtPlugin(
+    id = "OrtResult",
+    displayName = "ORT Result Reporter",
+    description = "A reporter that creates an ORT result file for the ORT run.",
```

Comment:
Nit: I guess the format is hard-coded to YAML? As you mention the JSON format for `RunStatistics`, should YAML be mentioned here as well?